### PR TITLE
feat: Allow merging nested values in plugin config secretRef

### LINKED ISSUE #1722: feat: Allow ApisixRoute's secretRef to provision nested configuration
### Description

Some plugins, like for example `aws-lambda` require to enter credentials in the ApisixRoute object:

```
      plugins:
        - name: aws-lambda
          config:
            authorization:
              iam:
                accesskey: "xxx"
                secretkey: "xxx"
```

We can fetch plugin configuration from secrets (proposed in #1408 and implemented in #1486) but current implementation does not support nested levels. 

I can use secretRef to set `config.accesskey` but not `config.authorization.iam.accesskey`.

Also, the secretRef in ApisixRoute behaves differently than secretRef in ApisixConsumer, where the secret [can be inserted at any point](https://apisix.apache.org/docs/ingress-controller/next/tutorials/enable-authentication-and-restriction/#key-auth-reference-secret-object).

I'd like to be able to set the nested plugin configuration or to use the secretRef in the same way I can do in ApisixConsumer.

### PR DESCRIPTION
**Pull request by Revolyssup**
### Type of change:
FIxes #1722 

This feature allows setting specific fields on plugin configuration by passing the path to the field via secrets in a dot separated way.

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [x] Documentation
- [x] Refactor
- [ ] Chore
- [x] CI/CD or Tests

### Pre-submission checklist:

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: docs/en/latest/concepts/apisix_route.md [hunk 1]
```diff
@@ -262,6 +262,46 @@ spec:
...
+You can also configure specific fields in the plugin configuration that are deeply nested by passing the path to that field. The path is dot-separated keys that lead to that field. The below example overrides the `X-Foo` header field in the plugin configuration from `v1` to `v2`.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+ #content is "v2"
+ name: echo
+data:
+ headers.X-Foo: djI=
```
Comment (kayx23) [inline docs/en/latest/concepts/apisix_route.md:276]: does this have to be base64?

Comment (Revolyssup) [reply docs/en/latest/concepts/apisix_route.md:276]: yes

File: docs/en/latest/concepts/apisix_route.md [hunk 2]
```diff
@@ -262,6 +262,46 @@ spec:
           secretRef: echo
 ```
 
+## Config with secretRef configuring nested fields
```
Comment (kayx23) [inline docs/en/latest/concepts/apisix_route.md:265]: Could you come up with a better title? Struggle to understand what this is trying to say

Comment (Revolyssup) [reply docs/en/latest/concepts/apisix_route.md]: @kayx23 Please check if current title is fine. I made the change/

File: pkg/providers/utils/insert_map.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	}
+
+	ikey := keys[0]
+	restKey := keys[1]
+	if dest[ikey] == nil {
+		dest[ikey] = make(map[string]interface{})
+	}
+	newDest, ok := dest[ikey].(map[string]interface{})
+	if !ok {
+		newDest = make(map[string]interface{})
```
Comment (AlinsRan) [inline pkg/providers/utils/insert_map.go:41]: dest[ikey] = newDest

Comment (Revolyssup) [reply pkg/providers/utils/insert_map.go:41]: thanks for catching this. I added more unit test cases to cover this scenario as well.

**Review by AlinsRan — APPROVED** [review]

**Review by monkeyDluffy6017 — APPROVED** [review]

**Pull request merged**